thirdparty/runtime_environments.go

```diff
@@ -0,0 +1,51 @@
+package thirdparty
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+type RuntimeEnvironmentsClient struct {
+	Client  *http.Client
+	BaseURL string
+	APIKey  string
+}
+
+// getImageNames returns a list of strings containing the names of all images from the runtime environments api
+// TODO: Remove nolint:unused when DEVPROD-6983 is resolved.
+//
+//nolint:unused
+func getImageNames(ctx context.Context, c *RuntimeEnvironmentsClient) ([]string, error) {
+	apiURL := fmt.Sprintf("%s/rest/api/v1/imageList", c.BaseURL)
+	request, err := http.NewRequest(http.MethodGet, apiURL, nil)
+	if err != nil {
+		return nil, err
+	}
+	request = request.WithContext(ctx)
+	request.Header.Add("Content-Type", "application/json")
+	request.Header.Add("api-key", c.APIKey)
+	result, err := c.Client.Do(request)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
```

Comment:
If `Do` returns a non-nil response, the body has to be closed per the [docs for `(Response).Body`](https://pkg.go.dev/net/http@go1.22.4#Response). Usually, we do this by just calling `defer resp.Body.Close()` to make sure we don't forget to close it.